sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -150,6 +195,39 @@ static az_result read_configuration_and_init_client()
   return AZ_OK;
 }
 
+// Read OS environment variables using stdlib function
+static az_result read_configuration_entry(
```

Comment:
I think we need to decide on extracting this as common code for all samples at some time.
/cc @ahsonkhan @gilbertw I think you were considering this before: was there any progress on adding a common way for samples to read from environment?